src/modules/SSHAuthorizedKeys.rb

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2016 SUSE LLC.
+#  All Rights Reserved.
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+#
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "yast"
+require "users/ssh_authorized_keyring"
+
+module Yast
+  # This module add support to handle SSH authorized keys.
+  #
+  # It's inteded to be a thin layer on top of SSHAuthorizedKeyring to be used by
+  # yast2-users module (which is mainly written in Perl).
+  class SSHAuthorizedKeysClass < Module
+    include Logger
+
+    attr_reader :keyring
+
+    def main
+      reset
+    end
+
+    # Read keys from a given home directory
+    #
+    # @see Yast::Users::SSHAuthorizedKeyring#read_keys
+    def read_keys(home)
+      keyring.read_keys(home)
+    end
+
+    # Write keys to a given home directory
+    #
+    # @see Yast::Users::SSHAuthorizedKeyring#write_keys
+    def write_keys(home)
+      keyring.write_keys(home)
+    rescue Users::SSHAuthorizedKeyring::HomeDoesNotExist => e
+      log.warn(e.message)
+      Report.Warning(
+        # TRANSLATORS: '%s' is a directory path
+        format(_("Home directory '%s' does not exist\n" \
+                 "so authorized keys will not be written."), e.directory)
+      )
+    rescue Users::SSHAuthorizedKeyring::NotRegularSSHDirectory => e
+      log.warn(e.message)
+      Report.Warning(
+        # TRANSLATORS: '%s' is a directory path
+        format(_("'%s' exists but it is not a directory. It may\n" \
+                 "a security issue so authorized keys will not\n" \
+                 "be written."), e.directory)
+      )
+    rescue Users::SSHAuthorizedKeyring::CouldNotCreateSSHDirectory => e
+      log.warn(e.message)
+      Report.Warning(
+        # TRANSLATORS: '%s' is a directory path
+        format(_("Could not create SSH directory under '%s',\n" \
```

Comment:
NP: Maybe instead of *Could not create SSH directory under '/home/foo'* display something like *Cannot create directory '/home/foo/.ssh'*, that's IMHO more clear about what exactly failed.

And actually you should reuse the already existing [Message.UnableToCreateDirectory](https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/Message.rb#L234) method, no need for a new translation.